Makefile

```diff
@@ -60,11 +60,16 @@ clean: FORCE
 dirclean: clean
 	rm -rf $(STAGING_DIR_HOST) $(STAGING_DIR_HOSTPKG) $(TOOLCHAIN_DIR) $(BUILD_DIR_BASE)/host $(BUILD_DIR_BASE)/hostpkg $(BUILD_DIR_TOOLCHAIN)
 	rm -rf $(TMP_DIR)
+ifneq ($(CONFIG_CCACHE),)
+ifeq ($(call qstrip,$(CONFIG_CCACHE_DIR)),)
```

Comment:
Can you explain this please?

Reply:
It only deletes the ccache directory, if ccache is configured and the user has not configured an explicit, likely external, ccache directory, with which we should not mess this destructively.
I was pondering, because of if destructiveness, deleting the ccache directory better belongs to the distclean target, which currently has a dependency on the cacheclean target.

Reply:
Makes sense to me. I'm wondering if we should print a warning message that "the external ccache" is not removed.